lib/libelastio-snap.c

```diff
@@ -10,8 +10,114 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <sys/ioctl.h>
+#include <string.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <dirent.h>
 #include "libelastio-snap.h"
 
+#define RELOAD_SCRIPT_PATH		"/etc/elastio/dla"
+
+#define BUF_SIZE				512
+#define RELOAD_SCRIPT_BDEV_SIZE	128
+#define RELOAD_SCRIPT_COW_SIZE	256
+
+struct reload_script_params {
+	int snapshot;
+	unsigned int minor;
+	unsigned int cache_size;
+	int ignore_snap_errors;
+	char bdev[RELOAD_SCRIPT_BDEV_SIZE];
+	char cow[RELOAD_SCRIPT_COW_SIZE];
+};
+
+int check_reload_dir()
+{
+	DIR *dir = opendir(RELOAD_SCRIPT_PATH);
+	if (dir) {
+		closedir(dir);
+		return 0;
+	}
+
+	return 1;
+}
+
+int elastio_snap_get_reload_params(unsigned int minor, struct reload_script_params *rp)
+{
+	FILE *fd;
+	int ret;
+	char buf[BUF_SIZE];
+	char f_name[BUF_SIZE];
+	char mode[32] = { 0 };
+	char ignore_errors[16] = { 0 };
+
+	snprintf(f_name, sizeof(f_name), "%s/reload_%d.sh", RELOAD_SCRIPT_PATH, minor);
+
+	fd = fopen(f_name, "r");
+	if (!fd)
+		return -1;
+
+	if (fread(buf, 1, BUF_SIZE, fd) == 0) {
+		fclose(fd);
+		return -1;
+	}
+
+	ret = sscanf(buf, "/usr/bin/elioctl reload-%s -c %u %s %s %u %s", mode, &rp->cache_size, rp->bdev, rp->cow, &rp->minor, ignore_errors);
```

Comment:
Yes, we see that installation scripts use exactly this path:

```
dist/initramfs/dracut/module-setup.sh:    inst /usr/bin/elioctl
dist/initramfs/initramfs-tools/hooks/elastio-snap:copy_exec /usr/bin/elioctl
dist/initramfs/initrd/boot-elastio-snap.sh:#%programs: /usr/bin/elioctl
```

 But I think it would be prettier defining this path above near `RELOAD_SCRIPT_PATH`